tasks/push-to-cdn/push-to-cdn.yaml

```diff
@@ -0,0 +1,140 @@
+        gwurl: $EXODUS_GW_URL
+        gwenv: $EXODUS_GW_ENV
+
+        logger: file:/proc/1/fd/1
+        loglevel: INFO
+
+        environments:
+        - prefix: exodus
+        EOF
+        echo "exodus-rsync.conf created here: $exodus_rsync_config_path"
+      # How to mount secrets that are in the vault here
+      # volumeMounts:
+      #   - name: secrets
+      #     mountPath: "/secrets"
+    - name: create-source-destination-for-files
+      image: quay.io/distribution_relengs/publish-to-dev-portal-test-image
+      script: |
+        #!/bin/bash
+        set -e  
+        BINARIES_PATH="$(workspaces.data.path)/$(params.binaries_dir)"
+
+        # Check if the directory exists
+        if [ ! -d "$BINARIES_PATH" ]; then
+            echo "Directory $BINARIES_PATH does not exist."
+            exit 1
+        fi
+
+        json_output="{}"
+        # Loop through each file in the directory
+        for BINARY_PATH in "$BINARIES_PATH"/*; do
+            if [ -f "$BINARY_PATH" ]; then
+                echo "Processing binary: $(basename "$BINARY_PATH")"
+                checksum=$(sha256sum "$BINARY_PATH" | awk '{print $1}')
+                first_two_chars=${checksum:0:2}
+                destination_path="exodus:/content/origin/files/sha256/$first_two_chars/$checksum/$(basename "$BINARY_PATH")"
+                echo "Destination dir: $destination_path"
+                echo ""
+                echo -n "$destination_path" >> "$(results.srcDestPath.path)"
+                # Add key:value (src:dest) pair to the json_output
+                json_output=$(echo "$json_output" | jq --arg key "$BINARY_PATH" --arg value "$destination_path" '.[$key]=$value')
+            fi
+        done
+
+        OUTPUT_PATH=$(params.subdirectory)/src_dest_combination.json
+        echo -n "$OUTPUT_PATH" > "$(results.destination_path.path)"
+        echo "$json_output" | tee "$(workspaces.data.path)/$OUTPUT_PATH"
+    - name: run-exodus-command
+    # TODO: Remove after testing
+      image: quay.io/distribution_relengs/publish-to-dev-portal-test-image
+      # Under construction
+      # TODO: Update this to do joined publish
+      # https://github.com/release-engineering/exodus-rsync?tab=readme-ov-file#joined-publish
+      script: |
+        #!/bin/bash
+        set -e       
+        conf_dir="$(workspaces.data.path)/$(params.subdirectory)/exodus-rsync.conf"
+        src_dest_json="$(workspaces.data.path)/$(params.subdirectory)/src_dest_combination.json"
+        cat "$src_dest_json" | jq -r 'to_entries[] | "\(.key) \(.value)"' | while read -r src dest; do
+            echo "Running exodus-rsync command..."
+            echo "exodus-rsync --exodus-conf $conf_dir -s $src -d $dest"
+            # TODO: Uncomment below to run the actual command
+            # exodus-rsync --exodus-conf "$conf_dir" "$src" "$dest"
+            echo "exodus-rsync command completed."
+            echo ""
+        done
+  # How to mount secrets that are in the vault here
```

Comment:
There's no Vault integration. That's only available in internal pipelines as Johnny mentioned. The secret would have to be defined somehow in the workspace and then you could access it by defining an environment variable. Something like the following:
```
valueFrom:
  secretKeyRef:
    name: secret
    key: key
```

Reply:
ty for the pointer, Searching for such instance in the repo, I see [create-github-release.yaml](https://github.com/konflux-ci/release-service-catalog/blob/bbecccc99e319143eb80de49be856b11e7198665/tasks/create-github-release/create-github-release.yaml#L49) and [create-pyxis-image.yaml](https://github.com/konflux-ci/release-service-catalog/blob/bbecccc99e319143eb80de49be856b11e7198665/tasks/create-pyxis-image/create-pyxis-image.yaml#L66) using similar env variables. 

I see they both are dependent on additional tasks, [collect-gh-params](https://github.com/konflux-ci/release-service-catalog/blob/bbecccc99e319143eb80de49be856b11e7198665/pipelines/release-to-github/release-to-github.yaml#L290) and [collect-pyxis-parms](https://github.com/konflux-ci/release-service-catalog/blob/bbecccc99e319143eb80de49be856b11e7198665/tasks/collect-pyxis-params/collect-pyxis-params.yaml) 
Is a separate task preferred here as well, something like `collect-exodus-params` ?

Reply:
This is because we have different secrets if we are running in stage or prod, and you can't determine what secret you want and then attach it to the pod in the same task. If this also will run against either the stage or prod instance, yes, we will need something like `collect-exodus-params` unfortunately

Reply:
I have created this MR for a new task - `prepare-exodus-params` (since there was bit more processing using _prepare_ instead of _collect_)  - https://github.com/konflux-ci/release-service-catalog/pull/455